syncer/dispatcher.go

```diff
@@ -25,8 +27,8 @@ func NewDispatcher(catchupSyncer syncer) *Dispatcher {
 	return &Dispatcher{
 		workQueue:     NewTargetQueue(),
 		catchupSyncer: catchupSyncer,
-		incoming:      make(chan Target, incomingBufferSize),
-		control:       make(chan interface{}),
+		incoming:      make(chan Target, IncomingBufferSize),
+		control:       make(chan interface{}, 1),
```

Comment:
The test was quite tricky and this solved my problem.  I'm not sure I could have tested this with a 0 buffer as before.  I think it is ok because it does not change much from a user uninterested in callback registration, but it opens up way simpler patterns for consumers (potentially non-test in the future) trying to synchronize with a callback.